### PR TITLE
Add new CI to test "make html"

### PR DESCRIPTION
**Pull request by mccalluc**
The linting PRs could conceivably impact the docs generation, but even apart from that, we'd like to be sure with each PR that we haven't broken anything, rather than wait for the deploy to check.

Dependencies are currently specified in the conda environment. Would it be possible to have a `requirement.txt` that is reused in different context?

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  docs build:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+          architecture: 'x64'
+      # TODO: Dependencies come from environment.yml.
+      # How should they be shared here? Install everything?
```
Comment (manzt) [inline .github/workflows/docs.yml:15]: Not sure of the best way to do this. I notice that some dependencies in `environment.yml` are repeated in `setup.py` as well, but know you are already thinking about breaking up dependencies with `extras_require`.

I think ideally `environment.yml` is used to setup a virtual environment with correct python version – maybe with some additional dependencies for running notebooks / building vitessce package (e.g., anndata, loompy, build). Development dependencies could then also be an `extras_requires` entry, importing contents from a standalone `requirements-dev.txt`, which can be pip installed here in isolation.

```bash
pip install -r requirements-dev.txt # just install dev dependencies (what to use in this workflow)
pip install -e .[dev] # install vitessce, dependencies, and dev dependencies
```

Maybe difficult to explain here. Happy to open a PR if interested.

**Review by manzt — APPROVED** [review]
Looks good to me. Left a comment regarding sharing dependencies but should be followed up in separate PR.

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: Docs build
+
+on: [push, pull_request]
+
+jobs:
+  docs build:
```
Comment (keller-mark) [inline .github/workflows/docs.yml:6]:
```suggestion
  docs_build:
```
I think this action is not running due to this space which is not allowed in the job name https://github.com/vitessce/vitessce-python/actions/runs/2155324608

File: .github/workflows/docs.yml (hunk 1 above)
Comment (mccalluc) [reply .github/workflows/docs.yml:15]: @manzt -- Yeah, if you can open up a follow-up PR, that'd be useful. Thanks.

**Comment by mccalluc** [comment]
failing with:
```
Running Sphinx v4.2.0
Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/sphinx/config.py", line 328, in eval_config_file
    exec(code, namespace)
  File "/home/runner/work/vitessce-python/vitessce-python/docs/conf.py", line 7, in <module>
    import vitessce
ModuleNotFoundError: No module named 'vitessce'
```

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+          architecture: 'x64'
+      # TODO: Dependencies come from environment.yml.
+      # How should they be shared here? Install everything?
+      - run: pip install sphinx==4.2.0 sphinx-rtd-theme==1.0.0 nbclean==0.3.2 nbsphinx==0.8.8
```
Comment (keller-mark) [inline .github/workflows/docs.yml:16]:
```suggestion
      - run: |
        pip install sphinx==4.2.0 sphinx-rtd-theme==1.0.0 nbclean==0.3.2 nbsphinx==0.8.8
        pip install -e .
```

The vitessce package defined in this repo needs to be installed into the environment before the docs are generated

**Review by keller-mark — CHANGES_REQUESTED** [review]

**Pull request merged**